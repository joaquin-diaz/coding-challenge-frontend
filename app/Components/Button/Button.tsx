import * as React from 'react';
import styled from '@emotion/styled';
import styles from '../../constants/styles';

const StyledButton = styled.button`
  background-color: ${styles.colors.primary};
  color: white;
  border: none;
  border-radius: 5px;
  outline: none;
  cursor: pointer;
  box-shadow: 0 1px 6px 0 rgba(32, 33, 36, 0.28);

  &:disabled {
    background-color: ${styles.colors.disabled};
    cursor: initial;
  }
`;

interface ButtonProps {
  className?: string;
  disabled?: boolean;
  type?: 'button' | 'submit' | 'reset';
}

const Button: React.FC<ButtonProps> = props => {
  return (
    <StyledButton className={props.className} disabled={props.disabled} type={props.type}>
      {props.children}
    </StyledButton>
  );
};

export default Button;
