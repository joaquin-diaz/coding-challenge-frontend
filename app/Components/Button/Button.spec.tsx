import * as React from 'react';

import Button from './Button';
import { mount } from 'enzyme';

describe('<Button />', () => {
  it('should render', () => {
    const wrapper = mount(<Button type="submit">Test</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});
