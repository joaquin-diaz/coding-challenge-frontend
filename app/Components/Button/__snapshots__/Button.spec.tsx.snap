// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render 1`] = `
.emotion-0 {
  background-color: #3f51b5;
  color: white;
  border: none;
  border-radius: 5px;
  outline: none;
  cursor: pointer;
  box-shadow: 0 1px 6px 0 rgba(32,33,36,0.28);
}

.emotion-0:disabled {
  background-color: #E1E2E1;
  cursor: initial;
}

<Button
  type="submit"
>
  <Styled(button)
    type="submit"
  >
    <button
      className="emotion-0"
      type="submit"
    >
      Test
    </button>
  </Styled(button)>
</Button>
`;
